src/targets/gpu/hip.cpp

```diff
@@ -293,6 +302,12 @@ argument get_preallocation(context& ctx, const std::string& id)
     return ctx.get_current_device().preallocations.at(id);
 }
 
+void gpu_fill(context& ctx, const argument& dst, int value)
+{
+    // TODO: Handle non-packed tensor when value is not 0
```

Comment:
What's special about value 0 for non-packed tensor such that it would work ? I thought it wouldn't work even for the zero.

Reply:
Yea it wont work for zero either.

Reply:
Because its writing a float type but using an integer value. So a value of `1` as `int` wont be a value of `1` as a `float`.

Reply:
> It writes Int value

It does write the int value, but if the type is float we dont want an in value of 1, we want a float value of 1(which has a different byte representation).

Reply:
Would be nice to include some verify tests